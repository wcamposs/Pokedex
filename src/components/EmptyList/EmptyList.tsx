import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

import { FontAwesome5 } from '@expo/vector-icons';

function EmptyList() {
	return (
		<View style={styles.container}>
			<FontAwesome5 name="frown" size={32} color="black" />
			<Text style={styles.text}>We couldn't find pokemons</Text>
		</View>
	);
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		marginTop: '50%',
		justifyContent: 'center',
		alignItems: 'center',
	},
	text: {
		marginVertical: 16,
		fontSize: 20,
		fontWeight: 'normal',
	},
});

export default EmptyList;
