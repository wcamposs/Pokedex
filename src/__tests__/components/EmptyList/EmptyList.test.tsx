import React from 'react';
import renderer from 'react-test-renderer';

import EmptyList from '../../../components/EmptyList/EmptyList';

jest.useFakeTimers();

describe('<EmptyList />', () => {
	it('renders empty list component correctly', () => {
		const tree = renderer.create(<EmptyList />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
