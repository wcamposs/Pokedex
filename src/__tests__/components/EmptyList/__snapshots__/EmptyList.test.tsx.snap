// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyList /> renders empty list component correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": "50%",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "normal",
        "marginVertical": 16,
      }
    }
  >
    We couldn't find pokemons
  </Text>
</View>
`;
